### DISCUSSION TITLE
Failed to resolve component: If this is a native custom element, make sure...

### DISCUSSION THREAD

**Discussion by ElYaiko** [nuxt/nuxt#17022] [Questions]
I'm trying to use `datatables.net-vue3` but I'm having errors when trying to import it as client or server component, my code is:

```
// plugins/datatable.client.ts
import DataTable from 'datatables.net-vue3'

export default defineNuxtPlugin(nuxtApp => {
  nuxtApp.vueApp.component('DataTable', DataTable)
})
```

```
// nuxt.config.js
...
build: {
  transpile: ['datatables.net-vue3'],
},
```

```
<template>
  <div>
    <ClientOnly>
      <DataTable class="display">
        <thead>
          <tr>
            <th>First</th>
            <th>Second</th>
          </tr>
        </thead>
      </DataTable>
    </ClientOnly>
  </div>
</template>
```

I get this error when trying to import it as client side component but the component loads almost ok:

```
[Vue warn]: Failed to resolve component: DataTable
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
```

And when triying to import it without "ClientOnly" I get 500 Error:

```
[h3] [unhandled] H3Error: Invalid value used as weak map key
    at createError (file:///Users/yonier/Sites/nube/node_modules/h3/dist/index.mjs:238:15)
    at Server.nodeHandler (file:///Users/yonier/Sites/nube/node_modules/h3/dist/index.mjs:428:21) {
  statusCode: 500,
  fatal: false,
  unhandled: true,
  statusMessage: 'Internal Server Error'
}
[nuxt] [request error] Invalid value used as weak map key
```

I'm newbie trying to understand how Nuxt3 works, I see this package datatables.net-vue3 has ".ssr" support... but I'm still getting errors.
Can somebody help me understand what I'm doing wrong? Thanks.

**Comment by sisou** [comment]
Same situation for me. I am registering a component via client-side plugin and using it inside `<ClientOnly>` just like you, but still get this warning. Excluding the component from resolution via the suggested config option makes the component not show up at all.

I get that it's just a warning, but it shows up when doing it just the way it is documented (if `nuxtApp.vueApp.component()` is the correct way to register the component?).

**Reply by danielroe** [reply]
You can enable `experimental.treeshakeClientOnly`. (Otherwise the component is still attempted to be rendered on server, it's just not included in the HTML.) See https://github.com/nuxt/framework/pull/5750 for more info.
